### ISSUE TITLE
calling .login(proof-value) when already having a session will not generate a new proof

### ISSUE THREAD

**Issue by NKCSS** [worldwide-asset-exchange/waxjs#222]
I encountered this while building the [example application for my Labs Grant](https://labs.wax.io/proposals/196). If you would call .login(different-proof-from-initial-login), the call will just return the logged in wallet, without showing a dialogue. The .proof property will also be available, but will have the original proof, not using the nonce that was passed to the login function.

**Comment by includenull** [comment]
https://github.com/worldwide-asset-exchange/waxjs/blob/develop/src/index.ts#L166-L172
```typescript
public async login(nonce?: string): Promise<string> {
  if (!this.user) {
    this.receiveLogin(await this.signingApi.login(nonce));
  }
   return this.user.account;
}
``` 

It looks like if there is a current user it will just return that.
Solution would be to call `WaxJS.logout()` first.

I'm not sure if changing this behaviour would break anything, needs some investigation.

**Comment by NKCSS** [comment]
Yeah, could be solved on either end; calling logout before calling login with nonce on the client would solve things, otherwise, you'd need to remember the nonce that was previously passed to login, compare when called again, and clear it on logout, etc.
